app/controllers/hub/state_file/efile_errors_controller.rb

```diff
@@ -4,7 +4,16 @@ class EfileErrorsController < Hub::StateFile::BaseController
     layout "hub"
 
     def index
-      @efile_errors = @efile_errors.where.not(service_type: "ctc").order(:source, :code)
+      order = [:source, :code]
+      if params[:sort_by].present?
+        order = [params[:sort_by]] + order
+      end
+
+      @efile_errors = @efile_errors.where.not(service_type: "ctc").order(*order)
+
+      if params[:filter_by_service_type].present?
```

Comment:
the only problem is that this is an operation on `@efile_errors` which is determined on line 12 using the sort order. so that change would probably look more like this?

```
@efile_errors = @efile_errors.where.not(service_type: "ctc")
 if params[:filter_by_service_type].present?
  @efile_errors = @efile_errors.where(service_type: params[:filter_by_service_type])
end

order = [:source, :code]
if params[:sort_by].present?
  order.prepend(params[:sort_by])
end

@efile_errors = @efile_errors.order(*order)
```

not sure if that is more or less readable tbh or if it's worth someone maybe following the pattern in future

Reply:
actually here's another possibility:

```
order = [:source, :code]
if params[:sort_by].present?
  order.prepend(params[:sort_by])
end

if params[:filter_by_service_type].present?
  filter = { service_type: params[:filter_by_service_type] }
end

@efile_errors = @efile_errors.where.not(service_type: "ctc").where(*filter).order(*order)
```

Reply:
fyi `@efile_errors` is actually created by cancancan's `load_and_authorize_resources`, which is why we can start this method with `@efile_errors = @efile_errors.where.not(service_type: "ctc")`, but i really like where you got to with this - thanks!

Reply:
well it errored and i don't think i have it in me to troubleshoot today so....i just undid it 😬